translator/translate/otel/common/eksdetector.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Detector interface {
+	getConfigMap(namespace string, name string) (map[string]string, error)
+}
+
+type EksDetector struct {
+	Clientset kubernetes.Interface
+}
+
+const (
+	authConfigNamespace = "kube-system"
+	authConfigConfigMap = "aws-auth"
+)
+
+var _ Detector = (*EksDetector)(nil)
+
+var (
+	detectorSingleton Detector
+	once              sync.Once
+)
+
+var (
+	getInClusterConfig  = func() (*rest.Config, error) { return rest.InClusterConfig() }
+	getKubernetesClient = func(confs *rest.Config) (kubernetes.Interface, error) { return kubernetes.NewForConfig(confs) }
+	// NewDetector creates a new singleton detector for EKS
+	NewDetector = func() (Detector, error) {
```

Comment:
This was done to make unit testing the singleton easier and it's exported for unit testing in the translators